internal/git/remote.go

```diff
@@ -0,0 +1,80 @@
+package git
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/evilmartians/lefthook/internal/log"
+	"github.com/spf13/afero"
+)
+
+var defaultRemotePath = mustGetDefaultRemotesDir()
+
+// mustGetDefaultRemotesDir returns the default directory for the lefthook remotes.
+func mustGetDefaultRemotesDir() string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	return filepath.Join(homeDir, ".lefthook-remotes")
+}
+
+// InitRemote clones or pulls the latest changes for a git repository that was specified as
+// a remote config repository. If successful, the path to the root of the repository will be
+// returned.
+func InitRemote(fs afero.Fs, url, rev string) (string, error) {
+	err := fs.MkdirAll(defaultRemotePath, 0755)
+	if err != nil && !errors.Is(err, os.ErrExist) {
+		return "", err
+	}
+
+	root := getRemoteDir(url)
+
+	_, err = fs.Stat(root)
+	if err == nil {
+		if err := updateRemote(fs, root, url, rev); err != nil {
+			return "", err
+		}
+		return root, nil
+	}
+
+	if err := cloneRemote(fs, root, url, rev); err != nil {
+		return "", err
+	}
+	return root, nil
+}
```

Comment:
I am in favor of having the `--sync-remotes/-s` flags re-sync the remote config. An alternative short version could be `-S` to avoid reserving `-s`. This could be useful later on since `-s` is commonly used as an alternative to `--silent` (althought `LEFTHOOK=0` already accomplishes this so it is most likely unnecessary).

Reply:
Feel free to use either `-S` or `-s`. Both variants work for me :)